src/NServiceBus.Core/EndpointConfiguration.cs

```diff
@@ -218,7 +218,9 @@ internal InitializableEndpoint Build()
                 Settings.SetDefault("PublicReturnAddress", publicReturnAddress);
             }
 
-            Settings.SetDefault<Conventions>(conventionsBuilder.Conventions);
+            var conventions = conventionsBuilder.Conventions;
+            Settings.SetDefault<Conventions>(conventions);
+            Settings.SetDefault("MessageTypes", new Lazy<Type[]>(() => Settings.GetAvailableTypes().Where(t => conventions.IsMessageType(t)).ToArray()));
```

Comment:
Why even use a Lazy here? We know we're going to invoke it at least one time, so is there any value in doing that using a lazy?